core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -144,25 +162,30 @@ public JoinDefinition define(AutoJoinerContext context) {
   public void prepareRun(BatchJoinerContext context) {
     if (conf.getNumPartitions() != null) {
       context.setNumPartitions(conf.getNumPartitions());
+      if (conf.getDistributionFactor() != null && conf.getDistributionFactor() < conf.getNumPartitions()) {
+        LOG.warn(String.format(
+          "Number of partitions (currently %d) should be greater than or equal to distribution factor (currently %d) "
```

Comment:
nit: remove the 'currently' in parentheses to shorten the message a little.

Reply:
done